### DISCUSSION TITLE
dash-uploader 0.7.0 pre-release(s)

### DISCUSSION THREAD

**Discussion by fohrloop** [np-8/dash-uploader#88] [General]
I have uploaded to PyPI dash-uploader version 0.7.0-a1. As this is a pre-release version, one must add the `--pre` flag when installing with pip. E.g.: 

```
python -m pip install dash-uploader --pre
```

The release introduces some fixes and changes, and some of the changes are **backwards incompatible**. For full list of changes, see the [CHANGELOG](https://github.com/np-8/dash-uploader/blob/dev/docs/CHANGELOG.md), and for upgrading instructions see the [Migration Guide](https://github.com/np-8/dash-uploader/wiki/Migration-Guide#060--070)

The most important things are:
- Multi-file upload is now fully supported
- @du.callback syntax changed, and `du.UploadStatus` objects were introduced.
- resumable.js was changed to flow.js
- Some CSS related changes

## Any comments?
Please, *feel free to comment* on this thread how you like the 0.7.0-a1 (or other prereleases)! I would be especially interested to know what kind of other changes should be included in the release of 0.7.0. Now it is a good time to collect also backwards incompatible changes to one release. One example could be styling/CSS related updates.

- The list for changes in 0.7.0 is also shown [here](https://github.com/np-8/dash-uploader/discussions/67) and it seems that it is close to be completed.
- [Here](https://github.com/np-8/dash-uploader/milestone/2) is a list of Issues with Milestone of 0.7.0

*I would be also interested if you have tried 0.7.0-a1 (or other pre-release) and verified that it is working in your use case!*

**Comment by vertical-space** [comment]
Hi, this is a really good addition to dash. Thanks for all your hard work!

I got version 0.7.0-a1 to run perfectly on my laptop on python 3.9.13 but when I tried to run it on one of our servers (python 3.7.11) I get the following error: 

Error while loading Python API: cannot import name 'final' from 'typing' (/opt/python/3.7.11/lib/python3.7/typing.py)

I was able to run version 0.5 no problem but not the latest prerelease version.

Cheers,

**Reply by vertical-space** [reply]
I found a workaround: by commenting out the line **from typing import final** from utils.py. Seems it wasn't required.

**Reply by fohrloop** [reply]
Thanks for the report! This should be an easy fix then.

**Comment by vertical-space** [comment]
Hi, just a couple of ideas:

It would be great if the cancel button had an id, so we could activate it via a callback (by incrementing n_clicks), eg if a user exceeded their file quota I would like to cancel the current upload and disable further uploading until they free up some space. 

Alternatively, could there be auto-disabling of the uploader if the downloads folder reaches a certain size (perhaps through the configure_upload step, but it would need to check the actual size of the downloads directory, not just rely on files processed by the uploader in the current session). 

Lastly, when the uploader completes, I would prefer if it didn't automatically add the name of the last uploaded file to the text_completed field, but this currently happens by default. (In fact, the use of text_completed could be optional altogether).

Cheers,

**Reply by fohrloop** [reply]
Thank you for taking time to suggest these! I'll have to take a better look on these when I have some tlme.

**Comment by AlexisRangelC** [comment]
Hi there, i got the next error while i'm using the latest version in pypi and the version 0.7.0

"POST /API/dash-uploader HTTP/1.1" 404

**Comment by divya-thota** [comment]
Hi, this is a great upgrade. 

I have a few suggestions,
1. When uploading a large file, it is getting stuck. I had to pause and resume multiple times for it to continue.
2. The filetypes restriction no longer works

I am currently using version 0.7.0-a1

**Comment by zorjo** [comment]
Hey guys, I want to deploy my dash app to the cloud and have it upload files directly to google drive. Any way to do this with dash-uploader? The filenames look nuts to me once I upload files!
